packages/datastore/datastore/src/storage/adapters/WebSQLAdapter.ts

```diff
@@ -110,35 +98,19 @@ export class WebSQLAdapter implements StorageAdapter {
     return this.readTransaction(`SELECT * FROM ${storeName}`, []);
   }
 
-  private getStoreNames(): Promise<string[]> {
-    return new Promise((resolve, reject) => {
-      const successCb = (tx: SQLTransaction, res: SQLResultSet) => {
-        const l = res.rows.length - 1;
-        const existingStores = [];
-        for (let i = 0; i < l; i++) {
-          const existingStore = res.rows.item(i);
-          existingStores.push(existingStore.name);
-        }
-        resolve(existingStores);
-      };
-      const errorCb = (tx: SQLTransaction, err: SQLError) => {
-        reject(err.message);
-        return false;
-      };
-      this.sqlite.transaction((tx) => {
-        tx.executeSql("SELECT name FROM sqlite_master WHERE name LIKE 'user_%'", [],
-          successCb,
-          errorCb
-        );
-      });
-    });
+  private async getStoreNames(): Promise<string[]> {
+    const res = await this.readTransaction(
+      "SELECT name FROM sqlite_master WHERE name LIKE 'user_%'",
+      []
+    );
+    return res.map((r) => r.name);
   }
 
-  private async transaction(query: string, args: any) {
+  private async transaction(query: string, args: any): Promise<any[]> {
     return new Promise((resolve, reject) => {
       const successCb = (tx: SQLTransaction, res: SQLResultSet) => {
         logger("Executed query with results", res.rows);
-        resolve(res.rows);
+        resolve(flattenResultSet(res.rows));
```

Comment:
Why this is needed?

Reply:
@wtrocki this was needed because the results came back as an SQLResult set object rather than an array of results. So I was getting errors when trying to use this.